includes/multi-currency/MultiCurrency.php

```diff
@@ -845,6 +845,54 @@ public function get_price( $price, string $type ): float {
 		return $this->get_adjusted_price( $converted_price, $apply_charm_pricing, $currency );
 	}
 
+	/**
+	 * Gets a raw converted amount based on the amount and currency codes passed.
+	 * This is a helper method for external conversions, if needed.
+	 *
+	 * @param float  $amount        The amount to be converted.
+	 * @param string $to_currency   The 3 letter currency code to convert the amount to.
+	 * @param string $from_currency The 3 letter currency code to convert the amount from.
+	 *
+	 * @return float The converted amount.
+	 *
+	 * @throws InvalidCurrencyException
+	 * @throws InvalidCurrencyRateException
+	 */
+	public function get_raw_conversion( float $amount, string $to_currency, string $from_currency = '' ): float {
+		$enabled_currencies = $this->get_enabled_currencies();
+
+		// If the from_currency is not set, use the store currency.
+		if ( '' === $from_currency ) {
+			$from_currency = $this->get_default_currency()->get_code();
+		}
+
+		// We throw an exception if either of the currencies are not enabled.
+		$to_currency   = strtoupper( $to_currency );
+		$from_currency = strtoupper( $from_currency );
+		foreach ( [ $to_currency, $from_currency ] as $code ) {
+			if ( ! isset( $enabled_currencies[ $code ] ) ) {
+				$message = 'Invalid currency passed to get_raw_conversion: ' . $code;
```

Comment:
I was thinking whether we can use early return for the validations checks? This may improve the readability and also avoid duplication. I am thinking of something like this:

```
foreach ( [ $to_currency, $from_currency ] as $code ) {
			if ( ! isset( $enabled_currencies[ $code ] ) ) {
				$this->handle_currency_error( 'Invalid currency' );
			}
}

...

// Validate the rate.
    if ( 0 >= $from_currency_rate ) {
        $this->handle_currency_error( 'Invalid rate' );
    }

...

private function handle_currency_error( string $message ): void {
    $log_message = 'Error in get_raw_conversion: ' . $message;
        
    Logger::error( $log_message );
    throw new InvalidCurrencyException( $log_message, 'wcpay_multi_currency_invalid_currency', 500 );
}
```

What do you think about this change?

Reply:
If we were going to create a method that handled logging and throwing the exception, it would need to be more generic so that it could be used for other instances where `InvalidCurrencyException` is thrown within the class. Something like this could work:

```php
	private function log_and_throw_invalid_currency_exception( $method, $currency_code, $code ) {
		$message = 'Invalid currency passed to ' . $method . ': ' . $currency_code;
		Logger::error( $message );
		throw new InvalidCurrencyException( $message, 'wcpay_multi_currency_invalid_currency', $code );
	}
```